modules/ebi-metagenomics/pimento/generatebcv/main.nf

```diff
@@ -38,44 +35,24 @@ process ASSESSMCPPROPORTIONS {
 
     if (strands == "") {
         """
-        touch ${prefix}_${var_region}_mcp_cons.tsv
+        touch ${assess_mcp_prop_prefix}_mcp_cons.tsv
 
         cat <<-END_VERSIONS > versions.yml
         "${task.process}":
-            mgnify-pipelines-toolkit: \$(get_mpt_version)
+            mi-pimento: \$( pimento --version | cut -d" " -f3 )
         END_VERSIONS
         """
-    } else if (library_check) {
+    } else {
         """
-        assess_mcp_proportions \\
+        pimento \\
+            gen_bcv \\
             -i ${fastq} \\
-            -s ${assess_mcp_prop_prefix} \\
             -st ${strands} \\
-            -o ./
-
-        library_strategy_check \\
-            -i ${library_check_input} \\
-            -s ${prefix} \\
-            -o ./
-
-        check_out=\$(cat ${prefix}_library_check_out.txt)
-
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            mgnify-pipelines-toolkit: \$(get_mpt_version)
-        END_VERSIONS
-        """
-    }  else {
-        """
-        assess_mcp_proportions \\
-            -i ${fastq} \\
-            -s ${assess_mcp_prop_prefix} \\
-            -st ${strands} \\
-            -o ./
+            -o ${assess_mcp_prop_prefix}
 
         cat <<-END_VERSIONS > versions.yml
         "${task.process}":
-            mgnify-pipelines-toolkit: \$(get_mpt_version)
+            mi-pimento: \$( pimento --version | cut -d" " -f3 )
```

Comment:
Maybe you can do version outside if-else? I guess there is no need to add it in if and in else

and here remove mi- from pimento as well